### PR TITLE
HDDS-10397. Restrict legacy bucket directory deletion through sh command.

### PR DESCRIPTION
**Pull request by ashishkumar50**
## What changes were proposed in this pull request?
When enableFileSystemPath is true and bucket layout is legacy, during key delete operation in normalizekey(prexecute) it removes trailing slash, but in DB key is present with trailing slash. And because of this KEY_NOT_FOUND error comes.
But fixing legacy bucket directory deletion using sh can lead to other issues. In this PR we are restricting legacy bucket directory deletion using sh command.
More discussion in [PR](https://github.com/apache/ozone/pull/6218).

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10397

## How was this patch tested?

New Integration test.

### CODE REVIEW

**Comment by ashishkumar50** [comment]
@sadanand48 Please help to review.

**Comment by adoroszlai** [comment]
@ashishkumar50 was CI run cancelled intentionally?

**Comment by ashishkumar50** [comment]
> @ashishkumar50 was CI run cancelled intentionally?

Yes, I thought of after review let it run. It's already passed in my fork branch.
I have run it now.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java [hunk 1]
```diff
@@ -1984,6 +1986,48 @@ public void testLinkedAndNonLinkedBucketMetaData()
...
+    String fileSystemEnable = generateSetConfString(OMConfigKeys.OZONE_OM_ENABLE_FILESYSTEM_PATHS, "true");
+    // Delete dirPath key, it should fail
+    args = new String[] {fileSystemEnable, "key", "delete", dirPath};
+    execute(ozoneShell, args);
+
+    // Check number of keys
+    args = new String[] {"key", "list", "o3://" + omServiceId + OZONE_URI_DELIMITER +
+          volumeName + OZONE_URI_DELIMITER + bucketName};
+    out.reset();
+    execute(ozoneShell, args);
```
Comment (sadanand48) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java:2018]: The above 3 lines are not needed as we are already checking key list size  below.

Comment (ashishkumar50) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java]: Done

**Review by sadanand48 — APPROVED** [review]
Thanks @ashishkumar50 for the patch, LGTM.

**Pull request merged**